src/main/flags.sh

```diff
@@ -4,19 +4,170 @@
+            # This can be removed when we transition away from the ocurl otel collector
+            #old: was just '--copt -static --copy -no-pie' before we dependended on libcurl
+            FLAGS=()
+            CCLIB=("-lssl" "-lcrypto" "-lz")
+            CCOPT=("-static" "-no-pie")
+        else
+            FLAGS=()
+            CCLIB=()
+            CCOPT=()
+        fi
+        ;;
+    # In general, we want to statically link to make it easier to distribute
+    # binaries since they won't carry runtime dependencies on various
+    # libraries which would need to be separately installed or managed.
+    # Additionally, this can be a bit fragile depending on which paths the
+    # loader uses, and where dylibs are expected to be (and how possible
+    # differences in the CI build versus other environments).
+    #
+    # However, on MacOS, it is a bit difficult to ensure libraries are linked
+    # statically. It is not possible to link all libraries statically.
+    # Notably, libSystem cannot (officially) be statically linked against,
+    # since MacOS does not have a stable syscall ABI, but rather relies on
+    # libSystem being stable. See, e.g.,
+    # - <https://developer.apple.com/forums/thread/706419>
+    # - <https://developer.apple.com/library/archive/qa/qa1118/_index.html>
+    # - `man 1 ld`
+    #
+    # As a result, we cannot use a similar strategy to our Alpine build (which
+    # works for Linux, since it is a rare case with a stable syscall ABI, See
+    # <https://unix.stackexchange.com/questions/473137/do-other-unix-like-kernels-have-stable-syscall-abis>).
+    # Previously we relied on a workaround where we deleted the `.dylib`
+    # versions of libraries, leaving only the `.a` versions, thus forcing them
+    # to be linked statically. However, this breaks the build in the case
+    # where later build steps require programs which dynamically linked
+    # against those libraries. Notably, this occurs for PCRE2. See
+    # <https://github.com/semgrep/semgrep/pull/10064>.
+    #
+    # One might hope that we could specify the linker to use the `.a` version
+    # without deleting the `.dylib`, but the only way to do this is to disable
+    # autolinking and specify each library individually. Otherwise, on MacOS,
+    # if both are present, the linker will link dynamically regardless.
+    #
+    # If you add a dependency which should be statically linked, it should be
+    # added to this list. If it has a `.pc` file, ensure that the library is
+    # appropriately added to PKG_CONFIG_PATH and use pkg-config to pick it up
+    # below (see, e.g., gmp). If we expect it to always be installed via brew,
+    # then use `brew --prefix <BREW PACKAGE NAME>`.
+    macosx)
+        # TODO: it would be best if this could be easily autogenerated from
+        # languages/
+        LANGS=(
+            "bash"
+            "c_sharp"
+            "c"
+            "cairo"
```

Comment:
What happens if you remove one?
We get an error in CI?
What happens it tomorrow we add a new language? Will the error be good enough to point to
this file?

Reply:
You would get an error in CI and locally. A representative error would be:
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make minimal-build
dune build _build/install/default/bin/semgrep-core
Generating flags for OS macosx
File "src/main/dune", line 2, characters 8-12:
2 |  (names Main)
            ^^^^
Undefined symbols for architecture arm64:
  "_octs_create_parser_go", referenced from:
      _camlTree_sitter_go__Parse__parse_source_string_328 in tree_sitter_go.a[4](tree_sitter_go__Parse.o)
      _camlTree_sitter_go__Parse__data_begin in tree_sitter_go.a[4](tree_sitter_go__Parse.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
File "caml_startup", line 1:
Error: Error during linking (exit code 1)
make[1]: *** [minimal-build] Error 1
make: *** [core] Error 2
```
(removing go)

Not quite good enough (imo) in terms of pointing to this file, but obvious and reproducible between CI and locally.

Reply:
since the error mention src/main/dune
you can probably add a comment in it saying "if you get a linking error like XXX, you probably need to modify .../flags.sh  and add the necessary library

Reply:
Added a printout from Make in 5a250d5b519220912a55ec234fddb427ed3f5441